### DISCUSSION TITLE
What do you think about FastStream?

### DISCUSSION THREAD

**Discussion by Lancetnik** [encode/broadcaster#120] [Q&A]
Hello there! You project looks great, but should you implement its own features if we already have the project implements the features you want to have:

* Serialization / deserialization to support broadcasting structured data
* A backend for RabbitMQ (and more in the the future)
* Add support for history for backends which provide persistence. (Redis Streams, Apache Kafka)
* Support for pattern subscribes in backends that support it

You can just build your backends on top of [FastStream](https://github.com/airtai/faststream) Brokers and use its features to provide users with your great API. I think it can make a sence.

**Comment by alex-oleshkevich** [comment]
They serve for different purposes. While the broadcaster is a simple library, FastStream looks like a higher-level application framework and does much more things.
They don't compete, but broadcaster could be a foundation for FastStream as Starlette did for FastAPI.

**Reply by Lancetnik** [reply]
>  broadcaster could be a foundation for FastStream as Starlette did for FastAPI

Rather the opposite - **FastStream** is just a fat client for brokers, but **broadcaster** is a final usecase with limited API and scope. So, **broadcaster** can use **FastStream** inside for its' very specific functional, but **FastStream** itself doesn't limit users by usecase as lower-level API.
